docs/agent/config/v3.mdx

```diff
@@ -462,11 +249,387 @@ These are a list of specifiers for what `Host` headers will be allowed to make r
 | `example.com`        |         | a hostname without preceding period will match `Host` header exactly (`example.com`)                                                     |
 | `.example.com`       |         | a hostname with a preceding period `Host` header suffix (`sub.example.com` or `foo.example.com`)                                         |
 
-##### Allow an IP address and a domain as `Host` headers
+##### Example
 
-```yaml
+Allow an IP address and a domain as `Host` headers:
+
+```yaml {2-4}
 agent:
   web_allow_hosts:
     - 8.8.8.8
     - example.com
 ```
+
+## Endpoint Definitions
+
+The `endpoints` field enables you to define and configure multiple endpoints.
+
+Defining multiple endpoints this way enables you to start pre-configured endpoints
+by name without having to memorize the right arguments every time through the
+`ngrok start` command. You can also use this field to start multiple endpoints at
+the same time from a single agent using the `ngrok start --all` flag.
+
+The endpoints field accepts a list of endpoint configurations, the list of available
+endpoint configuration options can be found here.
+
+### Example
+
+In the snippet below we are defining two endpoints named 'httpbin' and 'demo':
+
+```yaml
+endpoints:
+  - name: httpbin
+    url: https://alan-httpbin.ngrok.dev
+    upstream:
+      url: 8080
+  - name: demo
+    url: https://demo.inconshreveable.com
+    upstream:
+      url: 8181
+```
+
+#### Start the endpoint named 'httpbin'
+
+```sh
+ngrok start httpbin
+```
+
+#### Start all endpoints
+
+```sh
+ngrok start --all
+```
+
+### Endpoint Configuration Options
+
+| Name                                                 | Required | Protocols   | Description                                                                                                                                                            |
+| ---------------------------------------------------- | -------- | ----------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [`name`](#name)                                      | Required |             | A unique name for this endpoint's configuration.                                                                                                                       |
+| [`url`](#url)                                        | Required |             | The address you would like to use to forward traffic to your upstream service.                                                                                         |
```

Comment:
Okay, I might _also_ have mislead you here as well. You _can_ omit the `url` for the endpoint (and you'll receive a random one)

eg

```
version: 3 
agent:
  authtoken: PSYCH!
endpoints:
  - name: heyThisCanBeEmptyButIt'llBeAHTTPSEndpoint
    upstream:
      url: 2323

```